scrapers/csci_topics_scraper/main.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+# Python standard library
+import asyncio
+from operator import itemgetter
+import os
+import re
+import json
+import sys
+
+# External dependnecies
+import aiohttp
+import bs4
+import requests
+
+BeautifulSoup = lambda data: bs4.BeautifulSoup(data, features="lxml")
+
+
+async def get_topics_txts():
+    global session
+    url = "https://www.cs.rpi.edu/~goldsd/docs/"
+    async with session.get(url) as request:
+        soup = BeautifulSoup(await request.text())
+        links = soup.find("table").findAll("a")
+        out = []
+        for link in links:
+            if "topics-courses" in link["href"]:
+                out.append(link["href"])
+        # sorts reverse chronologically
+        return sorted(out, reverse=True, key=lambda x: (x[1:3] + str.lower(x[0])))
```

Comment:
Won't this order the semesters for a given year as `uXX`, `sXX`, `fXX` instead of `fXX`, `uXX`, `sXX` since it's doing lexicographical sorting for the spring/summer/fall modifier?

Reply:
It orders them `uXX-fXX`, `sXX`, which is the desired order. The summer and fall are in the same file.